### ISSUE TITLE
Run multiple nimbus-eth1 mainnet instances

### ISSUE THREAD

**Issue by tersec** [status-im/infra-nimbus#193]
Initially, these don't have to have validators attached to them, but function as a fourth backing EL in addition to Nethermind, Erigon, and Geth.

To facilitate syncing, it can be provided by a combination of era file syncing and/or a pre-prepared database synced close to current mainnet head.

**Comment by arnetheduck** [comment]
10 instances each for mainnet / holesky / sepolia - the database takes 2-3 weeks to create, so we'll pre-seed the nodes with a pre-prepared database copy

each instance needs about 300gb disk for the state - we should also think about setting it up in such a way that they have access to era1/era stores for historical block data

**Comment by jakubgs** [comment]
From a conversation with Jacek we can start with a setup like this and then grow from there:

- `mainnet` - Extend storage on all 7 hosts and add one `nimbus-eth1` node on each host, attached to all BNs on it.
- `sepolia` - Extend storage on the host and add four `nimbus-eth1` nodes, for each of the BNs on it.
- `holesky` - Replace Erigon EL nodes with `nimbus-eth` nodes on all 10 `erigon-01` hosts.

The priority is on deploying `nimbus-eth1` nodes on `mainnet` network. first.

**Comment by yakimant** [comment]
`nimbus.mainnet` has enough space, I will put it to the `/docker` volume together with `geth`.

```
❯ ansible -i ansible/inventory/test nimbus-mainnet-metal -a 'df -h /data /docker' -f1
linux-01.ih-eu-mda1.nimbus.mainnet | CHANGED | rc=0 >>
Filesystem      Size  Used Avail Use% Mounted on
/dev/sdb        2.9T  1.5T  1.4T  52% /data
/dev/sdc        3.5T  1.4T  1.9T  43% /docker
linux-02.ih-eu-mda1.nimbus.mainnet | CHANGED | rc=0 >>
Filesystem      Size  Used Avail Use% Mounted on
/dev/sdb        2.9T  1.5T  1.3T  55% /data
/dev/sdc        3.5T  1.4T  1.9T  43% /docker
linux-03.ih-eu-mda1.nimbus.mainnet | CHANGED | rc=0 >>
Filesystem      Size  Used Avail Use% Mounted on
/dev/sdb        2.9T  950G  1.8T  35% /data
/dev/sdc        3.5T  1.4T  1.9T  43% /docker
linux-04.ih-eu-mda1.nimbus.mainnet | CHANGED | rc=0 >>
Filesystem      Size  Used Avail Use% Mounted on
/dev/sdb        2.9T  1.1T  1.8T  38% /data
/dev/sdc        3.5T  1.4T  1.9T  43% /docker
linux-05.ih-eu-mda1.nimbus.mainnet | CHANGED | rc=0 >>
Filesystem      Size  Used Avail Use% Mounted on
/dev/sdb        2.9T  943G  1.8T  34% /data
/dev/sdc        3.5T  1.4T  1.9T  43% /docker
linux-06.ih-eu-mda1.nimbus.mainnet | CHANGED | rc=0 >>
Filesystem      Size  Used Avail Use% Mounted on
/dev/sdb        2.9T  946G  1.8T  34% /data
/dev/sdc        3.5T  1.4T  1.9T  43% /docker
linux-07.ih-eu-mda1.nimbus.mainnet | CHANGED | rc=0 >>
Filesystem      Size  Used Avail Use% Mounted on
/dev/sdb        2.9T  1.1T  1.7T  41% /data
/dev/sdc        3.5T  1.4T  2.0T  41% /docker
```

**Comment by yakimant** [comment]
`nimbus-eth1` is running on `linux-01.ih-eu-mda1.nimbus.mainnet` attached to it's beacon nodes.

Here is it's config template:
https://github.com/status-im/infra-role-nimbus-eth1/blob/master/templates/nimbus-eth1.service.j2

Looks like it needs some additional configuration in regards of syncing (prepared database or era files).

Found other config options here:
https://github.com/status-im/nimbus-eth1/blob/master/nimbus/config.nim

**Comment by yakimant** [comment]
We have those era files at the host:
```
❯ ls -1 /data/era/
mainnet-00000-4b363db9.era
...
mainnet-01198-7fa25a94.era
```

Shall I point `nimbus-eth1` to it with `--era-dir /data/era`?

**Comment by yakimant** [comment]
FYI

Beacon node EL stats:
<img width="1360" alt="Screenshot 2024-08-28 at 17 27 27" src="https://github.com/user-attachments/assets/8d16df9e-37fc-4fbc-bcea-3ed002f31f1a">

Errors in `nimbus-eth1` logs:
```
DBG 2024-08-28 15:25:36.018+00:00 Discovery send failed                      topics="eth p2p discovery" msg="(97) Address family not supported by protocol"
...
ERR 2024-08-28 15:26:39.042+00:00 Unexpected exception in rlpxAccept         topics="eth p2p rlpx" exc=EthP2PError err="Eth handshake for different network"
...
WRN 2024-08-28 15:27:32.303+00:00 Error while handling RLPx message          topics="eth p2p rlpx" peer=Node[37.24.131.128:30306] msg=newBlockHashes err="block announcements disallowed"
...
ERR 2024-08-28 15:28:23.082+00:00 Unexpected exception in rlpxAccept         topics="eth p2p rlpx" exc=EthP2PError err="Eth handshake for different network"
...
WRN 2024-08-28 15:28:29.446+00:00 Error while handling RLPx message          topics="eth p2p rlpx" peer=Node[136.244.57.56:30345] msg=newBlock err="block broadcasts disallowed"
```

**Comment by arnetheduck** [comment]
> Errors in nimbus-eth1 logs:

cc @mjfh can you take a look at this?

see https://github.com/status-im/nimbus-eth2/blob/unstable/docs/logging.md for our logging levels - in particular, remote nodes doing strange things should never result in any logs above debug level - from the point of view of nimbus, it is "normal" for remote nodes to misbehave and we should have logic in place that deals with the misbehavior rather than raising the issue to the user via logs - ie these are expected conditions, that there exist nodes that do strange things so they are not errors, warnings or even info.

**Comment by yakimant** [comment]
Exporting era1 can be done like that:

```
sudo geth --datadir=/docker/geth-mainnet/node/data  --mainnet export-history /docker/era1 0 15537393
```

where `15537393` is tha last block before merge.
See also:
- https://github.com/status-im/infra-nimbus/issues/189
- https://github.com/ethereum/trin/blob/master/e2store/README.md

**Comment by yakimant** [comment]
Shortcut for era1 files suggested by Jacek:
https://era1.ethportal.net

**Comment by yakimant** [comment]
RPC API doesn't show much:
```
❯ ./rpc.sh eth_syncing
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": {
    "startingBlock": "0x0",
    "currentBlock": "0x0",
    "highestBlock": "0x0"
  }
}
```

**Comment by tersec** [comment]
Syntactically, this is a valid, if minimalistic, response: https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_syncing

https://github.com/status-im/nimbus-eth1/blob/178d77ab310a79f3fa3a350d3546b607145a6aab/nimbus/core/chain/forked_chain.nim#L356-365 sets `highestBlock`:
```nim
proc setHead(c: ForkedChainRef,
             headHash: Hash256,
             number: BlockNumber) =
  # TODO: db.setHead should not read from db anymore
  # all canonical chain marking
  # should be done from here.
  discard c.db.setHead(headHash)

  # update global syncHighest
  c.com.syncHighest = number
```
but https://github.com/status-im/nimbus-eth1/blob/master/nimbus/nimbus_import.nim never calls `setHead(...)`.

`startingBlock` is arguably correct.

`currentBlock` is internally `syncCurrent`, and is updated from `nimbus import` by its `persistBlocks(...)` call, but these `syncCurrent`/`syncHighest`/syncStart` variables basically only reflect the syncing happening at that time, nothing per se the `nimbus import` command did. When Nimbus is run after the `nimbus import`, those are simply never changed from their defaults, because no syncing is happening.

However, it reports syncing because
```nim
  server.rpc("eth_syncing") do() -> SyncingStatus:
    ## Returns SyncObject or false when not syncing.
    # TODO: make sure we are not syncing
    # when we reach the recent block
    let numPeers = node.peerPool.connectedNodes.len
    if numPeers > 0:
      var sync = SyncObject(
        startingBlock: w3Qty com.syncStart,
        currentBlock : w3Qty com.syncCurrent,
        highestBlock : w3Qty com.syncHighest
      )
      result = SyncingStatus(syncing: true, syncObject: sync)
    else:
      result = SyncingStatus(syncing: false)
```
which isn't really correct. Having peers does not imply syncing.

So the issue is basically that it's not syncing, but falsely showing that it is syncing.

Its not syncing when connected to an EL is itself a bug but right now expected, known issue being addressed. I'm not sure I've seen the falsely-showing-syncing reported before.

**Comment by tersec** [comment]
https://github.com/status-im/nimbus-eth1/issues/2618

**Comment by tersec** [comment]
https://github.com/status-im/nimbus-eth1/pull/2619

**Comment by yakimant** [comment]
Progress
- geth full syncing: 9.9M of 20.7M (48%), 10d 19h
- eth1 era/era1 import: 6.4M of 20.7M (31%), 22h

**Comment by yakimant** [comment]
Stopped era/era1 import (would take 10 days).
And used state files from Jacek.

After starting the node, those appear in the logs:
```
DBG 2024-09-16 12:43:17.912+00:00 No finalized block stored in database, reverting to base
INF 2024-09-16 12:43:17.912+00:00 Database initialized                       base="(E7955DAE5D39FACC2C01EB33C3609F791E9938B70D0B32FD50873DD902DD3E00, 20475504)" finalized="(E7955DAE5D39FACC2C01EB33C3609F791E9938B70D0B32FD50873DD902DD3E00, 20475504)" head="(E7955DAE5D39FACC2C01EB33C3609F791E9938B70D0B32FD50873DD902DD3E00, 20475504)"
INF 2024-09-16 12:43:17.913+00:00 RLPx listener up                           topics="eth p2p" self=enode://f1ed95f5c159f1a20d358d60632b742e8ce2e432c69dc370bed303736d68577c2517be15c868d5a218883cf4186de9ef4953c7f5acd09dcacfa97b6aeeb113da@194.33.40.70:30304
WRN 2024-09-16 12:43:17.913+00:00 Engine API disabled, the node will not respond to consensus client updates (enable with `--engine-api`)
```

**Comment by yakimant** [comment]
BTW, this is how we run `nimbus-eth1` currently:

```sh
/docker/nimbus-eth1-mainnet-master/repo/build/nimbus \
    --network=mainnet \
    --data-dir='/docker/nimbus-eth1-mainnet-master/data/shared_mainnet_0' \
    --nat=extip:194.33.40.70 \
    --log-level=DEBUG \
    --listen-address=0.0.0.0 \
    --tcp-port=30304 \
    --udp-port=30304 \
    --max-peers=160 \
    --discovery=V4 \
    --jwt-secret=/docker/nimbus-eth1-mainnet-master/data/jwt.hex \
    --rpc=true \
    --ws=false \
    --graphql=false \
    --http-address=127.0.0.1 \
    --http-port=8546 \
    --rpc-api=eth,debug \
    --engine-api=false \
    --engine-api-ws=false \
    --metrics=true \
    --metrics-address=0.0.0.0 \
    --metrics-port=9401 \
    --era1-dir=/docker/era1 \
    --era-dir=/data/era
```

**Comment by yakimant** [comment]
```
❯ ./rpc.sh eth_blockNumber
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": "0x1386e70"
}

❯ ./rpc.sh eth_syncing
{
  "jsonrpc": "2.0",
  "id": 1,
  "result": false
}
```

**Comment by tersec** [comment]
```
    --engine-api=false \
    --engine-api-ws=false \
```
Turn these on/true

**Comment by yakimant** [comment]
Done!

BTW, more interesting messages:
```
WRN 2024-09-17 09:05:17.889+00:00 getUncles()                                topics="core_db" ommersHash=3EF3400295E8F4DFE6AED323009018E85DADBB1A4D960FC5CEE0709D484F6E97 error="KvtNotFound(Kvt, ctx=get, error=GetNotFound)"
```

**Comment by tersec** [comment]
> Done!
> 
> BTW, more interesting messages:
> 
> ```
> WRN 2024-09-17 09:05:17.889+00:00 getUncles()                                topics="core_db" ommersHash=3EF3400295E8F4DFE6AED323009018E85DADBB1A4D960FC5CEE0709D484F6E97 error="KvtNotFound(Kvt, ctx=get, error=GetNotFound)"
> ```

Do you know which block was involved here (if you have a block hash or number for example)? The issue is that only PoW blocks have uncles. PoS (post-merge) blocks don't. But, by that token, PoS blocks are supposed to have an ommersHash indicating this.

So the question is, is it a PoW block which might have real uncles and there's a bug finding them, or is the bug in detecting that a PoS block should not have uncles.

**Comment by yakimant** [comment]
Hm, I'm not sure how to help, but you can check the logs yourself:
`linux-01.ih-eu-mda1.nimbus.mainnet:/var/log/service/nimbus-eth1-mainnet-master/`

There are quite a lot of such warning: ~22k yesterday, ~9k today already.

**Comment by yakimant** [comment]
BTW, logs are on Kibana also:
https://nimbus-logs.infra.status.im/goto/022fa9f0-75a0-11ef-a1cd-13d81fb37e1c

**Comment by yakimant** [comment]
Changed a port for BNs to use engine port, should be working now.

Also made a PR for review: https://github.com/status-im/infra-nimbus/pull/199

**Comment by tersec** [comment]
> Hm, I'm not sure how to help, but you can check the logs yourself: `linux-01.ih-eu-mda1.nimbus.mainnet:/var/log/service/nimbus-eth1-mainnet-master/`
> 
> There are quite a lot of such warning: ~22k yesterday, ~9k today already.

Non-issue/overzealous logging: https://github.com/status-im/nimbus-eth1/issues/2639

**Comment by jakubgs** [comment]
After discussing the layout with Dustin and Jacek we came to a conclusion that setups where multiple BNs with one EL are unsupported and should be avoided. Instead we will run two ELs per host, and half of BNs will run without. Branches `unstable` and `stable` should run with an EL, and others should run without any(`--no-el`).

In addition to that we want EL diversity, so we'll start with a split similar to `nimbus.holesky` fleet:

* `linux-01.ih-eu-mda1.nimbus.mainnet` -> `neth1-01.ih-eu-mda1.nimbus.mainnet`
* `linux-02.ih-eu-mda1.nimbus.mainnet` -> `neth1-02.ih-eu-mda1.nimbus.mainnet`
* `linux-03.ih-eu-mda1.nimbus.mainnet` -> `neth1-03.ih-eu-mda1.nimbus.mainnet`
* `linux-04.ih-eu-mda1.nimbus.mainnet` -> `erigon-01.ih-eu-mda1.nimbus.mainnet`
* `linux-05.ih-eu-mda1.nimbus.mainnet` -> `erigon-02.ih-eu-mda1.nimbus.mainnet`
* `linux-06.ih-eu-mda1.nimbus.mainnet` -> `geth-01.ih-eu-mda1.nimbus.mainnet`
* `linux-07.ih-eu-mda1.nimbus.mainnet` -> `geth-02.ih-eu-mda1.nimbus.mainnet`

This probably should be implemented using a flag in the layout file, same way we enable validator clients with `vc = true`.

Is this correct @tersec ?

**Comment by yakimant** [comment]
@tersec need your review of new mainnet layout:
https://github.com/status-im/infra-nimbus/pull/206/files#diff-3f8ce3d2e84c759a7dd04d47c2e9fe41ba16bbbcf0098de7bf0bd7673945043e

It's changed considerably:
- stable, testing, 2x unstable -> stable, testing, unstable, libp2p (same as holesky)
- linux-07 had 4x libp2p before
- removed `extra_flags: {'debug-enable-yamux': true}`, not sure we need it
- shall I keep `public_api` on `geth` nodes?
- there are `max_peers: 10000` and `open_libp2p_ports: false`, do we need it?

**Comment by tersec** [comment]
> [@tersec](https://github.com/tersec) need your review of new mainnet layout: https://github.com/status-im/infra-nimbus/pull/206/files#diff-3f8ce3d2e84c759a7dd04d47c2e9fe41ba16bbbcf0098de7bf0bd7673945043e
> 
> It's changed considerably:
> 
>     * stable, testing, 2x unstable -> stable, testing, unstable, libp2p (same as holesky)
> 
>     * linux-07 had 4x libp2p before
> 
>     * removed `extra_flags: {'debug-enable-yamux': true}`, not sure we need it
> 
>     * shall I keep `public_api` on `geth` nodes?
> 
>     * there are `max_peers: 10000` and `open_libp2p_ports: false`, do we need it?

- We don't need `--debug-enable-yamux`, no: https://github.com/ethereum/consensus-specs/pull/3866#issuecomment-2276195113 and we don't have QUIC support yet, though that would have been worth testing if so

- stable, testing, 2x unstable -> stable, testing, unstable, libp2p (same as holesky) is a useful change

- 4x libp2p was probably overkill. maybe there was a reason, but sure.

- I can't speak to the exact motivation on the very high peer count, but it's a good thing to test that it doesn't break on one node.

- I'm not sure what `open_libp2p_ports` does. It's not directly a Nimbus flag.

- We need some public API, ideally on a non-optimistic (i.e. has a functioning EL, so Geth/Nethermind/Erigon, not yet), e.g., to support https://eth-clients.github.io/checkpoint-sync-endpoints/

**Comment by jakubgs** [comment]
I can speak to the `open_libp2p_ports` setting. It was something most probably requested by Zahary or Jacek. The intention was to not open LibP2P ports on the firewall to compare how well discovery handles it and how quickly nodes gain peers.

I've decided to drop it as nobody has talked about this in ages and it made the setup more complex.

**Comment by yakimant** [comment]
1. removed `open_libp2p_ports: false` to test close libp2p ports
2. Moved `unstable` `public_api` to beacon node with geth. What do we do with `tesging` `public_api`? We don't have a `tesing` BN with EL.
3. Moved testing max peers 10k on stable BN with geth

**Comment by jakubgs** [comment]
That's a good question, does the BN used for public API endpoint need to have an EL? If so we might have to make an exception.

**Comment by tersec** [comment]
> That's a good question, does the BN used for public API endpoint need to have an EL? If so we might have to make an exception.

It should, yes

**Comment by arnetheduck** [comment]
> how well discovery handles it and how quickly nodes gain peers.

this is still an important test that we should keep, since many of our users don't have a public ip and we want to catch regressions.

**Comment by yakimant** [comment]
Ok, I will keep this test.

Previously it was for unstable and testing nodes (different hosts) with geth EL attached.
Any preference for node now?
You can get an overview of fleet here:
https://github.com/status-im/infra-nimbus/blob/104a36a2f20ff1b5b73ead7069da7e89cca90d4c/ansible/vars/layout/mainnet.yml

**Comment by jakubgs** [comment]
> Ok, I will keep this test.

I think we can achieve it not by changing already existing rules but by adding extra rules that deny access for specific ports.

**Comment by yakimant** [comment]
Merged the PR:
https://github.com/status-im/infra-nimbus/pull/206

Layout is mostly changed.

What left:
- [ ] https://github.com/status-im/infra-nimbus/issues/211
- [ ] https://github.com/status-im/infra-nimbus/issues/212
- [ ] `nimbus-eth1` (aka `nel`) setup (era1, database copy)
- [ ] cleanup (nginx,..)

**Comment by yakimant** [comment]
> 10 instances each for mainnet / holesky / sepolia

I actully realised we need `holesky`/`sepolia` eth1 too.

For `holesky` we could take 1 of each geth/neth/erigon, which is 3x4=12 nodes:
https://github.com/status-im/infra-nimbus/blob/master/ansible/vars/layout/holesky.yml

For `sepolia1 maybe some nodes? Or we can have a new host(s) for eth1?
https://github.com/status-im/infra-nimbus/blob/master/ansible/vars/layout/sepolia.yml

@jakubgs, what do you think?

**Comment by jakubgs** [comment]
I don't think we need 10 for each network just yet, but we need to start. For holesky we could replicate the same thing as we did with Mainnet, so we have `geth`, `erigon` and `nimbus-eth1`, rather than `geth`, `erigon`, and `nethermind`.

My main worry with that is that we might not want to switch big Holesky hosts with lots of validators to `nimbus-eth1` just yet. Maybe we could just start with for example the first 5, since they only hold 40 validators, out of the 100k we run, so even if they break the impact would be minimal.

As for Sepolia, I would ask Dustin and Jacek what they want. Would they prefer to split our existing 100 Sepolia validators between two hosts, or should we just mix up the ELs on the `linux-01.ih-eu-mda1.nimbus.sepolia` host a bit.

**Comment by tersec** [comment]
We're genuinely looking to ramp this up. Sepolia doesn't have enough ELs for this, and we're like them on all the networks as well, including (without validators but then, none of the testnet mainnet fleet has validators) mainnet.

But yeah, it's fine to focus on part of the Sepolia host and some of the smaller Holesky nodes in terms of validators -- but there still several of those.

**Comment by jakubgs** [comment]
Right, and would replacing `nethermind` with `nimbus-eth1` in [our layout](https://github.com/status-im/infra-nimbus/blob/master/ansible/vars/layout/holesky.yml) not be enough on Holesky? That's 10 hosts with 4 EL nodes on each.

**Comment by tersec** [comment]
> Right, and would replacing `nethermind` with `nimbus-eth1` in [our layout](https://github.com/status-im/infra-nimbus/blob/master/ansible/vars/layout/holesky.yml) not be enough on Holesky? That's 10 hosts with 4 EL nodes on each.

Sure.

**Comment by yakimant** [comment]
update on mainnet sync status:

1. `nec-01:stable`: era imported, cache keys generation in progress
- era import took **11d6h**
2. `nec-02:stable`: db copied, cache keys generated, synced form network
- db copy from `20241204-f034af4`
- cache keys generation took **13h**
- network sync took **10d12h**
2. `nec-03:stable`: era imported, cache keys generation in progress
- era import from 6.5M took **11d16h**
3. `nec-01:unstable` & `nec-02:unstable`: stuck in sync, need to resync (era import or db copy)
4. `nec-03:unstable`: slowely syncing from network from scratch (**8.7M in 24d**)

**Comment by yakimant** [comment]
# Current `mainnet` issues (from my perpective)

## Slow network sync

8.7M blocks in 24d and still running (see `nec-03:unstable`)

## CL head growth is not gradual

I suppose it should be smooth, but it's like that (`nec-02:stable`):
<img width="697" alt="Image" src="https://github.com/user-attachments/assets/1dfd18e1-541c-444b-815b-744a0741837c" />

## ERA import is slow

I've heard it should take about 3 days, but it took 10, see `nec-02:stable`

## `nec-03:unstable` RPC timeouts (flaping)

See consul alerts in `#⁠dev-ops-alerts` discord

## Sync from network is slow

See `nec-02:stable`, it took 10d12h to sync from 21M to 21.6M. Notice the sync peers count much less than 8.

<img width="688" alt="Image" src="https://github.com/user-attachments/assets/b50803ae-2132-493e-a69b-a249fd0e8024" />

**Comment by mjfh** [comment]
I would suggest that the graphs are unfortunate to see what is really happening.

In the first `Sync` graph, the `beacon_head` (now ` nec_sync_head`) is the internal ceiling (or scrum target) which has not much general information. I pitch the `latest` (now `nec_execution_head`) against the `base` variable which shows the usable state (see `Base vs Latest`)

The `buddies` graph is most useful when pitched against the queue of block lists. The latter must decrease over time when imported into the database. If not, there is something wrong (see `Buddies vs Block Lists` below)

![Image](https://github.com/user-attachments/assets/fbca10b4-8616-4da7-b3e6-09cd96902df9)

Legend
>  _yellow (Base vs Latest) -- `latest` aka `nec_execution_head`_
>  _pink (Base vs Latest) -- `base`_
>  _blue (Buddies vs Block Lists) -- `buddies` aka `nec_sync_peers`_
>  _green (Buddies vs Block Lists) -- `nec_sync_block_lists_staged`_

Just to show the usefulness from my PoV:

As this graphs show, there is s problem starting at  ~1am where the _yellow_ graph has an uptick and drops immediately after (due to a request from the `CL`.) This leads to a syncer deadlock, apparently. The _green_ graph stays at 1 (block list entries) which implies that there are data to process which cannot be processed.

**Comment by yakimant** [comment]
@mjfh I've added those graphs as you suggested:
<img width="1367" alt="Image" src="https://github.com/user-attachments/assets/7a813753-a4cd-4f8b-9770-ed6a40300b83" />
for easier debug.

You can look at those here:
https://grafana.infra.status.im/d/O8C3xSRNk/nimbus-eth1

I still keep the `nec_consensus_head` vs `nec_execution_head` as those are showing, that node is in sync (when those 2 meet).

**Comment by mjfh** [comment]
`Buddies vs Block Lists` reveals a problem.

why:
>  The block lists queue can occasionally exceed 16 but that should quickly resolve. tis is a similar deadlock situation to what I described above.

A problem occurs withy the sync graph when the `sync_head` is far away from the `exec_head` -- I doubt the usefulness of a graph that spans 20m blocks and has its lower graph (the `exec_head`) increase by a speed of 2 blocks / sec. :) Maybe you can manage a gap in the ordinate (mixed log/linear scaling?)

**Comment by yakimant** [comment]
- holesky import took 27h
- sepolia still running

**Comment by yakimant** [comment]
I'm closing this issue.

Ther leftovers are either https://github.com/status-im/infra-nimbus/issues/212 or operational issues.

**Comment by yakimant** [comment]
just a small update on current state:
1. mainnet stopped syngind, nimbus guys are on it
2. hoodi is fine, also quick to sync, fresh network
3. holesky we dont care
4. sepolia - never managed to sync, so now running geth back
